tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -37,93 +43,107 @@ public static void OnInstrumentPublished(Instrument instrument, MeterListener li
         }
 
         bool shouldEnable;
-        var enabledMeterNames = Tracer.Instance.Settings.OpenTelemetryMeterNames;
+        var enabledMeterNames = _settings.OpenTelemetryMeterNames;
 
         if (enabledMeterNames.Length > 0)
         {
-            shouldEnable = enabledMeterNames.Contains(meterName);
+            shouldEnable = Array.Exists(enabledMeterNames, name => name.Equals(meterName, StringComparison.Ordinal));
         }
         else
         {
             shouldEnable = !meterName.StartsWith("System.", StringComparison.Ordinal);
         }
 
-        if (shouldEnable)
+        if (!shouldEnable)
         {
-            var state = CreateMetricState(instrument);
-            if (state != null)
-            {
-                listener.EnableMeasurementEvents(instrument, state);
-                Log.Debug("Enabled measurement events for instrument: {InstrumentName} from meter: {MeterName}", instrument.Name, meterName);
-            }
+            return;
+        }
+
+        var instrumentType = GetInstrumentType(instrument.GetType().FullName);
+        if (instrumentType == null)
+        {
+            Log.Debug("Skipping unsupported instrument: {InstrumentName} of type: {InstrumentType}", instrument.Name, instrument.GetType().FullName);
+            return;
+        }
+
+        var temporality = TemporalityFor(instrumentType.Value, _settings.OtlpMetricsTemporalityPreference);
+        var identity = new MetricStreamIdentity(instrument, instrumentType.Value);
```

Comment:
As a follow-up item, I think we should track adding properties for `MeterVersion` and `MeterTags` since Meters are also unique based on those fields (in addition to `MeterName`)